From: Juergen Borleis <jbe@pengutronix.de>
Date: Mon, 23 Nov 2015 11:09:49 +0100
Subject: [PATCH] Skov-DO63x: provide the most specific compatible string entry

Used to select the matching devicetree via bootspec and inside Barebox.
These specific strings are required to use one binary Barebox, one binary
Linux kernel and ony single root filesystem to support all boards.

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 arch/arm/boot/dts/imx6dl-skov-dol63x-revA.dts | 2 +-
 arch/arm/boot/dts/imx6dl-skov-dol63x-revB.dts | 2 +-
 arch/arm/boot/dts/imx6dl-skov-dol63x-revC.dts | 2 +-
 arch/arm/boot/dts/imx6q-skov-dol63x-revA.dts  | 2 +-
 arch/arm/boot/dts/imx6q-skov-dol63x-revB.dts  | 2 +-
 arch/arm/boot/dts/imx6q-skov-dol63x-revC.dts  | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl-skov-dol63x-revA.dts b/arch/arm/boot/dts/imx6dl-skov-dol63x-revA.dts
index a94e68d6fab9..16b753bd12a7 100644
--- a/arch/arm/boot/dts/imx6dl-skov-dol63x-revA.dts
+++ b/arch/arm/boot/dts/imx6dl-skov-dol63x-revA.dts
@@ -19,5 +19,5 @@
 
 / {
 	model = "Freescale i.MX6 SKOV Solo Board";
-	compatible = "fsl,imx6dl", "fsl,imx6dl_cpu";
+	compatible = "skov,dol63x-imxdl-revA", "skov,dol63x", "fsl,imx6dl";
 };
diff --git a/arch/arm/boot/dts/imx6dl-skov-dol63x-revB.dts b/arch/arm/boot/dts/imx6dl-skov-dol63x-revB.dts
index 38ea3d5ccd1d..0b5b03e7478c 100644
--- a/arch/arm/boot/dts/imx6dl-skov-dol63x-revB.dts
+++ b/arch/arm/boot/dts/imx6dl-skov-dol63x-revB.dts
@@ -19,5 +19,5 @@
 
 / {
 	model = "Freescale i.MX6 SKOV Solo Board";
-	compatible = "fsl,imx6dl", "fsl,imx6dl_cpu";
+	compatible = "skov,dol63x-imxdl-revB", "skov,dol63x", "fsl,imx6dl";
 };
diff --git a/arch/arm/boot/dts/imx6dl-skov-dol63x-revC.dts b/arch/arm/boot/dts/imx6dl-skov-dol63x-revC.dts
index 6e14835fe3c0..afa335b45844 100644
--- a/arch/arm/boot/dts/imx6dl-skov-dol63x-revC.dts
+++ b/arch/arm/boot/dts/imx6dl-skov-dol63x-revC.dts
@@ -19,5 +19,5 @@
 
 / {
 	model = "Freescale i.MX6 SKOV Solo Board";
-	compatible = "skov,dol63x", "fsl,imx6dl";
+	compatible = "skov,dol63x-imxdl-revC", "skov,dol63x", "fsl,imx6dl";
 };
diff --git a/arch/arm/boot/dts/imx6q-skov-dol63x-revA.dts b/arch/arm/boot/dts/imx6q-skov-dol63x-revA.dts
index 1d1b94774bd2..7c11304bf1d8 100644
--- a/arch/arm/boot/dts/imx6q-skov-dol63x-revA.dts
+++ b/arch/arm/boot/dts/imx6q-skov-dol63x-revA.dts
@@ -19,7 +19,7 @@
 
 / {
 	model = "Freescale i.MX6 SKOV Quad Board";
-	compatible = "fsl,imx6q", "fsl,imx6_cpu";
+	compatible = "skov,dol63x-imxq-revA", "skov,dol63x", "fsl,imx6q";
 };
 
 &sata {
diff --git a/arch/arm/boot/dts/imx6q-skov-dol63x-revB.dts b/arch/arm/boot/dts/imx6q-skov-dol63x-revB.dts
index 66c81513b527..f9be4f7d83aa 100644
--- a/arch/arm/boot/dts/imx6q-skov-dol63x-revB.dts
+++ b/arch/arm/boot/dts/imx6q-skov-dol63x-revB.dts
@@ -19,7 +19,7 @@
 
 / {
 	model = "Freescale i.MX6 SKOV Quad Board";
-	compatible = "fsl,imx6q", "fsl,imx6_cpu";
+	compatible = "skov,dol63x-imxq-revB", "skov,dol63x", "fsl,imx6q";
 };
 
 &sata {
diff --git a/arch/arm/boot/dts/imx6q-skov-dol63x-revC.dts b/arch/arm/boot/dts/imx6q-skov-dol63x-revC.dts
index ece4aa5acfda..629147555512 100644
--- a/arch/arm/boot/dts/imx6q-skov-dol63x-revC.dts
+++ b/arch/arm/boot/dts/imx6q-skov-dol63x-revC.dts
@@ -19,7 +19,7 @@
 
 / {
 	model = "Freescale i.MX6 SKOV Quad Board";
-	compatible = "skov,dol63x", "fsl,imx6q";
+	compatible = "skov,dol63x-imxq-revC", "skov,dol63x", "fsl,imx6q";
 };
 
 &sata {
